exp/addons/internal/controllers/clusterresourceset_helpers.go

```diff
@@ -146,6 +141,33 @@ func (r *ClusterResourceSetReconciler) getOrCreateClusterResourceSetBinding(ctx
 	return clusterResourceSetBinding, nil
 }
 
+// ensureOwnerRefs ensures Cluster and ClusterResourceSet owner references are set on the ClusterResourceSetBinding.
+func ensureOwnerRefs(clusterResourceSetBinding *addonsv1.ClusterResourceSetBinding, clusterResourceSet *addonsv1.ClusterResourceSet, cluster *clusterv1.Cluster) []metav1.OwnerReference {
+	clusterResourceSetBinding.Name = cluster.Name
+	clusterResourceSetBinding.Namespace = cluster.Namespace
+	clusterResourceSetBinding.OwnerReferences = util.EnsureOwnerRef(clusterResourceSetBinding.OwnerReferences, metav1.OwnerReference{
+		APIVersion: clusterv1.GroupVersion.String(),
+		Kind:       "Cluster",
+		Name:       cluster.Name,
+		UID:        cluster.UID,
+	})
+
+	clusterResourceSetBinding.OwnerReferences = util.EnsureOwnerRef(clusterResourceSetBinding.OwnerReferences, *metav1.NewControllerRef(clusterResourceSet, clusterResourceSet.GroupVersionKind()))
+
+	// Ensure only one of the OwnerReferences is set as the controller.
```

Comment:
I don't think there's any need to set controller: true for any of these ownerRefs as I don't think we're using it. Removing it and setting a basic ownerRef would simplify the logic here.

Reply:
+1 to remove this.